### ISSUE TITLE
[BUG] Forge Mod can't send type to plugin very well

### ISSUE THREAD

**Issue by martin1194** [bloodmc/GriefDefender#335]
Plugin: GriefDefender 1.5.4
Server: Magma-9e0b1b1 / mohist-1.12.2-148
MC: 1.12.2
MOD: industrialcraft-2-2.8.221-ex112.jar

https://griefdefender.github.io/debug/?lKmKC8OqcF (Magma)
https://griefdefender.github.io/debug/?tn6K8C0fhN (Sponge)
https://griefdefender.github.io/debug/?R5Mp9Qqkvb (Mohist)
I need thestate=type:generatorto ban the machine in plugin.
But it does not work very well.

In sponge
```
server=global, source=#any
source=minecraft:any, source=minecraft:player
state=facing:down, state=material:ic2_material_machine
state=transparent:false, state=type:generator
target=#any, target=ic2:any
target=ic2:te, used_hand=minecraft:main_hand
used_item=ic2:te, world=world
```

In Magma
```
server=global, source=#any
source=minecraft:any, source=minecraft:player
state=facing:down, state=material:field_151573_f
state=transparent:false, state=type:invalid
target=#any, target=ic2:any
target=ic2:te, world=world
```

In Mohist
```
server=global, source=#any
source=minecraft:any, source=minecraft:player
state=facing:down, state=material:field_151573_f
state=transparent:false, state=type:invalid
target=#any, target=ic2:any
target=ic2:te, world=world
```

**Comment by bloodmc** [comment]
This is mohist/magma issue not GD.

**Comment by bloodmc** [comment]
I'll have a workaround for this issue in next dev build.

**Comment by martin1194** [comment]
Thank you so much :3

**Comment by bloodmc** [comment]
https://griefdefender.github.io/debug/?njT1Ku3e0k

Targets should show up properly now. I just need to see if I can add a workaround for item usage too.

**Comment by martin1194** [comment]
I'm sorry. It seems ic:te still can't show metadata or the type.

bloodmc <notifications@github.com> 於 2020年12月24日 週四 下午12:21寫道：

**Comment by martin1194** [comment]
here is the detail.
It can't use material:field_151573_f at flag cuz there is very many ic:te
is material:field_151573_f.

[image: image.png]

martin chan <martinc1194@gmail.com> 於 2020年12月24日 週四 下午12:28寫道：

**Comment by bloodmc** [comment]
You won't need to use material or state. I'll be using the TileEntity ID instead.

**Comment by martin1194** [comment]
fixed in griefdefender-bukkit-1.5.5-DEV7
